cores/esp32/esp32-hal-gpio.c

```diff
@@ -104,14 +104,14 @@ extern void ARDUINO_ISR_ATTR __pinMode(uint8_t pin, uint8_t mode)
 #endif
 
     if (pin >= SOC_GPIO_PIN_COUNT) {
-        log_e("Invalid pin selected");
+        log_e("Invalid pin (%i) selected", pin);
         return;
     }
 
     if(perimanGetPinBus(pin, ESP32_BUS_TYPE_GPIO) == NULL){
         perimanSetBusDeinit(ESP32_BUS_TYPE_GPIO, gpioDetachBus);
         if(!perimanClearPinBus(pin)){
-            log_e("Deinit of previous bus failed");
+            log_e("Deinit of previous bus from GPIO %i failed", pin);
```

Comment:
Pin is not set as GPIO yet, so better to use "pin" instead of "GPIO".

Reply:
Pin: usually the physical pin number on a chip or module
GPIO: usually the hardware IO+IOBUS number inside the chip. Do not apply for special purpose IOs
IO: usually wider sense, which includes GPIO and special purpose IOs

BUT because we are special and have IO mode = GPIO and all our IOs can be GPIOs, let's use IO from now on for when we want to specify something on the hardware level that might not be in GPIO mode.

If we are in `pinMode` or going through the matrix, then we are in GPIO mode. If we are using something analog (ADC, DAC, Touch, etc.) we are in RTC mode. Else we are in special mode (I'm pretty sure we don't currently use it, but UART0 starts that way, Flash/PSRAM are that way also and if IO numbers match SPI, we could use it there too for better 80MHz support)